pkg/integrations/cloudwatch_exporter/cloudwatch_exporter.go

```diff
@@ -43,13 +51,19 @@ func (e *exporter) MetricsHandler() (http.Handler, error) {
 	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		e.logger.Debug("Running collect in cloudwatch_exporter")
 
+		// since we have called refresh, we have loaded all the credentials
+		// into the clients and it is now safe to call concurrently. Defer the
+		// clearing, so we always clear credentials before the next scrape
+		e.cachingClientFactory.Refresh()
+		defer e.cachingClientFactory.Clear()
```

Comment:
This matches what was previously being done in YACE itself https://github.com/nerdswords/yet-another-cloudwatch-exporter/blob/v0.51.0/pkg/job/scrape.go#L32